### DISCUSSION TITLE
.select() throwing errors when called again

### DISCUSSION THREAD

**Discussion by rdmchr** [supabase/supabase#1629] [Questions] [answered]
Im using Vue + Vuex and when my page is loaded it stores all the information in the Vuex store. Then if I create a new entry to the database using my app its supposed to refresh the data using the same Vuex action, but instead I get these two errors:

```
[Vue warn]: Unhandled error during execution of native event handler 
  at <Anonymous onClick=fn<bound addMemory> > 
  at <Anonymous> 
  at <IonPage isInOutlet=true registerIonPage=fn<registerIonPage> > 
  at <CreateMemory ref=Ref< 
Proxy { <target>: {…}, <handler>: {…} }
 > key="/tabs/createMemory" isInOutlet=true  ... > 
  at <IonRouterOutlet tabs=true > 
  at <IonTabs> 
  at <IonPage isInOutlet=true registerIonPage=fn<registerIonPage> > 
  at <Tabs ref=Ref< 
Proxy { <target>: {…}, <handler>: {…} }
 > key="/tabs/memoriesList" isInOutlet=true  ... > 
  at <IonRouterOutlet> 
  at <IonApp> 
  at <App> runtime-core.esm-bundler.js:38
```
```
Uncaught (in promise) TypeError: Body not allowed for GET or HEAD requests
    Request browser-ponyfill.js:356
    fetch browser-ponyfill.js:456
    fetch browser-ponyfill.js:455
    then types.js:29
    Babel 4
    fetchAllMemories index.ts:75
    wrappedActionHandler vuex.esm-browser.js:815
    dispatch vuex.esm-browser.js:488
    boundDispatch vuex.esm-browser.js:373
    _callee$ CreateMemory.vue:88
    tryCatch runtime.js:63
    invoke runtime.js:293
    defineIteratorMethods runtime.js:118
    Babel 6
    addMemory CreateMemory.vue:88
    patchStopImmediatePropagation runtime-dom.esm-bundler.js:347
    callWithErrorHandling runtime-core.esm-bundler.js:154
    callWithAsyncErrorHandling runtime-core.esm-bundler.js:163
    callWithAsyncErrorHandling runtime-core.esm-bundler.js:173
    invoker runtime-dom.esm-bundler.js:333
    addEventListener runtime-dom.esm-bundler.js:283
    patchEvent runtime-dom.esm-bundler.js:301
    patchProp runtime-dom.esm-bundler.js:369
    mountElement runtime-core.esm-bundler.js:3905
    processElement runtime-core.esm-bundler.js:3868
    patch runtime-core.esm-bundler.js:3788
    componentEffect runtime-core.esm-bundler.js:4298
    reactiveEffect reactivity.esm-bundler.js:42
    effect reactivity.esm-bundler.js:17
    setupRenderEffect runtime-core.esm-bundler.js:4263
    mountComponent runtime-core.esm-bundler.js:4222
    processComponent runtime-core.esm-bundler.js:4182
    patch runtime-core.esm-bundler.js:3791
    mountChildren runtime-core.esm-bundler.js:3975
    mountElement runtime-core.esm-bundler.js:3896
    processElement runtime-core.esm-bundler.js:3868
    patch runtime-core.esm-bundler.js:3788
    mountChildren runtime-core.esm-bundler.js:3975
    mountElement runtime-core.esm-bundler.js:3896
    processElement runtime-core.esm-bundler.js:3868
    patch runtime-core.esm-bundler.js:3788
    componentEffect runtime-core.esm-bundler.js:4298
    reactiveEffect reactivity.esm-bundler.js:42
    effect reactivity.esm-bundler.js:17
    setupRenderEffect runtime-core.esm-bundler.js:4263
    mountComponent runtime-core.esm-bundler.js:4222
    processComponent runtime-core.esm-bundler.js:4182
    patch runtime-core.esm-bundler.js:3791
    mountChildren runtime-core.esm-bundler.js:3975
    mountElement runtime-core.esm-bundler.js:3896
```
This is the call that seems to produce the errors:
```JS
const memories = await memoriesDatabase.select(`id, memory, date, title, tags(id, tag, color)`).limit(limit);
```
And I'm triggering that method from my component using this code:
```JS
await this.$store.dispatch('fetchAllMemories', 10);
```
And I have confirmed that the limit is set correctly. Also as stated above the error does not occur the first time the method is called.
I would prefer to not subscribe to the database since that wouldn't be necessary since there won't be any updates to listen for except from the user itself, meaning that I could just trigger a refresh of the data. Is it possible to do so?

**Comment by inian** [comment, marked as answer]
Can you post a screenshot of the browser network panel here with the request that is failing @rdmchr? It looks like a GET (which doesn't allow for request body) and POST request are being conflated

**Reply by rdmchr** [reply]
![image](https://user-images.githubusercontent.com/66096031/118984828-2dc78380-b97e-11eb-99ad-db886cc879ac.png)
That's all I get when I trigger the method.

**Reply by inian** [reply]
Hmm looks like the request which fails is not even being sent to Supabase. 
Another random idea - can you use `supabase.from('memories')` each time instead of caching it in the `memoriesDatabase` variable?

**Reply by rdmchr** [reply]
Huh, that worked. Thank you very much. But I don't quite understand why. Should I just use this longer form or is there something I can do to go back to the way it was before without messing it up?

**Reply by inian** [reply]
oh nice. Each `supabase.from(memories)` is to construct a new request chain. And the request is sent when you `await` it like `await  memories.select()`. When the second request re-uses the same initial chain, it errors out cos the request has already been sent - something like that I am guessing.

**Reply by rdmchr** [reply]
That's interesting and there is no close() call or something similar to prevent that, right?

**Reply by inian** [reply]
I have been thinking about an explicit `close` call too, but for now the request is sent when you do the `await`
